### PR TITLE
DT-1786 Add Create Audit Event endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/integration/endtoend/AuditTest.kt [hunk 1]
```diff
@@ -68,13 +61,54 @@ class AuditTest {
...
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(BodyInserters.fromValue(AuditEvent(what = "basicAuditEvent")))
+        .exchange()
+        .expectStatus().isCreated
+
+      AuditService.log.info(auditRepository.hashCode().toString())
+
+      await untilCallTo { mockingDetails(auditRepository).invocations.size } matches { it == 1 }
+
+      verify(telemetryClient).trackEvent(eq("hmpps-audit"), com.nhaarman.mockitokotlin2.any(), isNull())
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/integration/endtoend/AuditTest.kt:84]: It looks like you're using the original mockito `any` in this class too - I think you only need the nhaarman version?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListenerTest.kt [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
-@ActiveProfiles("test")
 internal class HMPPSAuditListenerTest {
-  @MockBean
-  private lateinit var auditService: AuditService
-
-  @Autowired
-  private lateinit var listener: HMPPSAuditListener
+  private val auditService: AuditService = mock()
+  private val listener: HMPPSAuditListener =
+    HMPPSAuditListener(auditService = auditService, ObjectMapper().registerModule(JavaTimeModule()))
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListenerTest.kt:15]: I was just checking to see how the object mapper is configured in the main application - is there an issue with indentation in the jackson config in `application.yml`?  If so it's a shame that wasn't caught by a test...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditService.kt [hunk 3]
```diff
@@ -32,11 +46,15 @@ class AuditService(private val telemetryClient: TelemetryClient, private val aud
 
   fun sendAuditEvent(auditEvent: AuditEvent) {
     log.debug("Audit {} ", auditEvent)
+    auditMessagingTemplate.send(
+      queueName,
+      MessageBuilder.withPayload(mapper.writeValueAsString(auditEvent)).build()
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/services/AuditService.kt:51]: Will add this into next PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListenerTest.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/listeners/HMPPSAuditListenerTest.kt:15]: Have reverted to integrationtest to allow spring to inject the objectmapper directly - this is a better test to use the same config of the mapper when running the application

**Pull request merged**